aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -15,220 +15,443 @@ uid: fundamentals/servers/kestrel
+
+**ConfigureHttpsDefaults(Action<HttpsConnectionAdapterOptions>)**  
+Specifies a configuration `Action` to run for each HTTPS endpoint. Calling `ConfigureHttpsDefaults` multiple times replaces prior `Action`s with the last `Action` specified.
+
+**Configure(IConfiguration)**  
+Creates a configuration loader for setting up Kestrel that takes an [IConfiguration](/dotnet/api/microsoft.extensions.configuration.iconfiguration) as input. The configuration must be scoped to the configuration section for Kestrel.
+
+**ListenOptions.UseHttps**  
+Configure Kestrel to use HTTPS.
+
+`ListenOptions.UseHttps` extensions:
+
+* `UseHttps` &ndash; Configure Kestrel to use HTTPS with the default certificate. Throws an exception if no default certificate is configured.
+* `UseHttps(string fileName)`
+* `UseHttps(string fileName, string password)`
+* `UseHttps(string fileName, string password, Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(StoreName storeName, string subject)`
+* `UseHttps(StoreName storeName, string subject, bool allowInvalid)`
+* `UseHttps(StoreName storeName, string subject, bool allowInvalid, StoreLocation location)`
+* `UseHttps(StoreName storeName, string subject, bool allowInvalid, StoreLocation location, Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(X509Certificate2 serverCertificate)`
+* `UseHttps(X509Certificate2 serverCertificate, Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(HttpsConnectionAdapterOptions httpsOptions)`
+
+`ListenOptions.UseHttps` parameters:
+
+* `filename` is the path and file name of a certificate file, relative to the directory that contains the app's content files.
+* `password` is the password required to access the X.509 certificate data.
+* `configureOptions` is an `Action` to configure the `HttpsConnectionAdapterOptions`. Returns the `ListenOptions`.
+* `storeName` is the certificate store from which to load the certificate.
+* `subject` is the subject name for the certificate.
+* `allowInvalid` indicates if invalid certificates should be considered, such as self-signed certificates.
+* `location` is the store location to load the certificate from.
+* `serverCertificate` is the X.509 certificate.
+* `httpsOptions` are options to configure HTTPS.
+
+In production, HTTPS must be explicitly configured. At a minimum, a default certificate must be provided.
+
+Supported configurations described next:
+
+* No configuration
+* Replace the default certificate from configuration
+* Change the defaults in code
+
+*No configuration*
+
+Kestrel listens on `http://localhost:5000` and `https://localhost:5001` (if a default cert is available).
+
+Specify URLs using the:
+
+* `ASPNETCORE_URLS` environment variable.
+* `--urls` command-line argument.
+* `urls` host configuration key.
```

Comment:
it's important to distinguish that this must come from the host configuration, not the app configuration. There's an FAQ where people try to set Urls in appsettings.json and it doesn't work because appsettings.json isn't read until after the host is initialized. https://github.com/aspnet/MetaPackages/issues/247.  Kestrel's new config schema doesn't have this limitation, it reads directly from the config much later in the startup process.

Reply:
I'm going to address this under line 175 because that's where the 2.0 mention of the `urls` config key is covered.